### DISCUSSION TITLE
Got high confidence while the adoption rate was 0%.

### DISCUSSION THREAD

**Discussion by Masa-Shin** [whitesource/merge-confidence#16] [Q&A] [answered]
**What Renovate type, platform and version are you using?**

hosted App, Github.

**Describe the bug**

<img width="827" alt="スクリーンショット 2020-12-08 12 44 08" src="https://user-images.githubusercontent.com/34234442/102004891-3923dc80-3d58-11eb-9f37-335eb21d1fb0.png">

Renovate bot made a PR and in the summary there is an update whose adoption rate was 0%, passig rate was 100%, and confidence was `high`. I suppose it is not a valid estimate. 

**Relevant debug logs**

It happens at 5 days ago and the log is not available now.

**To Reproduce**

Install `@storybook/vue` package of version `6.0.28` by npm/yarn and run renovate bot.

**Additional context**

I suppose when adoption rate is exactly equals to 0%, passing rate should also be equals to 0%. Or if it was nearly 0% and got rounded, its confidence might not be so high.

**Comment by rarkins** [comment, marked as answer]
This is currently intentional. If a package is more than 3d old and passed all tests then we rate it as high confidence, meaning people should feel safe to upgrade to it. The maximum is "very high" confidence which is reached after higher adoption.

>  when adoption rate is exactly equals to 0%, passing rate should also be equals to 0%

No. Passing rate refers to test results. 100% of tests passed and is correct.

**Reply by Masa-Shin** [reply]
Thanks now I understand. Somehow I got a wrong idea that tests come after adoptions...

**Reply by rarkins** [reply]
No problem, we are aiming to document it better. Tests come first, and if they're very high and the package is > 3 days old then we will rate it as high confidence so that people begin upgrading. Once enough have upgraded and none have rolled back, it can elevate to very high confidence for the majority to feel confident
